packages/material-ui-lab/src/DatePicker/DatePickerKeyboard.test.tsx

```diff
@@ -0,0 +1,223 @@
+import * as React from 'react';
+import { expect } from 'chai';
+import { spy } from 'sinon';
+import TextField from '@material-ui/core/TextField';
+import { fireEvent, screen, act } from 'test/utils';
+import { StaticDatePicker, DesktopDatePicker, DesktopDatePickerProps } from './DatePicker';
+import { createPickerRender, adapterToUse } from '../internal/pickers/test-utils';
+import { CalendarSkeleton } from '../CalendarSkeleton';
+import { MakeOptional } from '../internal/pickers/typings/helpers';
+
+function TestKeyboardDatePicker(
+  props: MakeOptional<DesktopDatePickerProps, 'value' | 'onChange' | 'renderInput'>,
+) {
+  const [value, setValue] = React.useState<unknown>(
+    props.value ?? new Date('2019-01-01T00:00:00.000'),
+  );
+
+  return (
+    <DesktopDatePicker
+      value={value}
+      onChange={(newDate) => setValue(newDate)}
+      renderInput={(props) => <TextField placeholder="10/10/2018" {...props} />}
+      {...props}
+    />
+  );
+}
+
+describe('<DatePicker /> keyboard interactions', () => {
+  const render = createPickerRender({ strict: false });
+
+  context('input', () => {
+    function typeInPicker(value: string) {
+      fireEvent.change(screen.getByRole('textbox'), {
+        target: { value },
+      });
+    }
+
+    it('Allows to change selected date from the input according to `format`', () => {
+      const onChangeMock = spy();
+      render(
+        <TestKeyboardDatePicker
+          onChange={onChangeMock}
+          inputFormat={adapterToUse.lib === 'moment' ? 'DD/MM/YYYY' : 'dd/MM/yyyy'}
+        />,
+      );
+
+      typeInPicker('10/11/2018');
+
+      expect(screen.getByRole('textbox')).to.have.value('10/11/2018');
+      expect(onChangeMock.callCount).to.equal(1);
+    });
+
+    it("Doesn't accept invalid date format", () => {
+      render(<TestKeyboardDatePicker />);
+
+      typeInPicker('01');
+      expect(screen.getByRole('textbox')).to.have.attr('aria-invalid', 'true');
```

Comment:
Please grep in the codebase to seek for what's most popular and use it, before introducing a new API. They are aliases: https://github.com/nathanboktae/chai-dom#attrname-value.

```suggestion
      expect(screen.getByRole('textbox')).to.have.attribute('aria-invalid', 'true');
```